### PR TITLE
bpf: egressgw: tolerate BPF_FIB_LKUP_RET_NO_NEIGH on older kernels

### PR DESCRIPTION
**Pull request by julianwiedmann**
https://github.com/cilium/cilium/pull/30128 recently clarified that EgressGW can't fall back to the neigh-map when using `fib_do_redirect()`. Instead we need to tolerate a `BPF_FIB_LKUP_RET_NO_NEIGH` lookup result on older kernels (pre-5.10) that don't have `bpf_redirect_neigh()`.

Have `egress_gw_fib_lookup_and_redirect()` catch this case, and just let the packet continue as it normally would:

- in `from-overlay`, the kernel stack will route it for us and do the neigh lookup.
- in `to-netdev`, we just exit on the current interface. Users that require the correct egress interface need to run a 5.10+ kernel.

### CODE REVIEW

**Review by ysksuzuki — APPROVED** [review]
LGTM, Thanks!

File: bpf/bpf_overlay.c [hunk 1]
```diff
@@ -414,8 +414,13 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx,
...
 
 			/* to-netdev@bpf_host handles SNAT, so no need to do it here. */
-			return egress_gw_fib_lookup_and_redirect(ctx, snat_addr,
-								 daddr, ext_err);
+			ret = egress_gw_fib_lookup_and_redirect(ctx, snat_addr,
+								daddr, ext_err);
+			if (ret != CTX_ACT_OK)
+				return ret;
+
+			if (!revalidate_data(ctx, &data, &data_end, &ip4))
```
Comment (jibi) [inline bpf/bpf_overlay.c:422]: why is this needed now?

Comment (julianwiedmann) [reply bpf/bpf_overlay.c:422]: The revalidation? Because we called `ipv4_l3()` earlier, and the code further below calls  `lookup_ip4_endpoint(ip4)`.

I'd actually want to reshuffle this section at a later time, and handle the local-delivery *first*. To the point where we can return `DROP_UNROUTABLE` after attempting all expected delivery options.

**Review by jibi — APPROVED** [review]
